programmers/문자열-내-p와-y의-개수.js

```diff
@@ -0,0 +1,9 @@
+function solution(s) {
+  let p = 0, y = 0;
+  s.split('').forEach((val) => {
+    val = val.toLowerCase()
+    if (val === 'p') p += 1;
+    if (val === 'y') y += 1;
+  })
```

Comment:
s를 split()하는 과정에서 이미 시간이 소요됩니다.
JS에서는 문자열도 배열처럼 순회할 수 있으므로 굳이 배열로 변환한뒤에 배열의 메서드를 사용할 필요는 없는 것이지요.
또한 각각의 원소(val)별로 toLowerCase()를 해주는것보다 하나의 문자열을 한번에 변환하는 것이 코드를 해석하기에 훨씬 좋습니다.
```suggestion
  const lowerStr = s.toLowerCase()
  for (let unit of lowerStr) {
    if (unit === 'p') p += 1;
    else if (unit === 'y') y += 1;
  }
```

Reply:
c style for문보다 함수형 foreach를 쓰는걸 지향하는 편인데 split이 아닌 [...str].foreach 방식을 사용해도 될까요?

Reply:
> c style for문보다 함수형 foreach를 쓰는걸 지향하는 편인데 split이 아닌 [...str].foreach 방식을 사용해도 될까요?
되긴 하나 똑같은 코드인데 한번 더 고민해보게 하는 코드입니다.
본인의 스타일은 잠시 내려두고 조금이라도 더 좋은 코드를 쫓아가 봅시다 :D
코드는 가장 심플하고 유딩도 알아볼 수 있게 짜는게 좋은거랍니다 ^^